backend/server.js

```diff
@@ -31,14 +33,60 @@ app.post('/user', UserController.createUser);
 app.get('/user/:id', UserController.findUserById);
 app.put('/user/:id', UserController.updateUserById);
 
-app.get('/songs', (req, res) => {
+app.get('/songs', async (req, res) => {
+  const spotifyId = req.query.spotifyId;
+  const { lastScraped, listenStats } = await UserController.directFindUserBySpotifyId(spotifyId)
+  console.log('lastScraped', lastScraped)
+  const songsSorted = Object.keys(listenStats).sort(function(a, b) {return -(listenStats[a] - listenStats[b])})
+  const topN = songsSorted.slice(0, TOP_N_SONGS_TO_SHOW)
+  console.log('topN', topN)
+
+  spotifyApi.setAccessToken(req.query.token);
+  spotifyApi
+    .getTracks(topN)
+    .then(
+      async (data) => {
+        // console.log(data.body.tracks)
+        const trackData = data.body.tracks.map(track => {
+          return {
+            ...track,
+            plays: listenStats[track.id]
+          }
+        })
+        res.send(trackData);
+      },
+      (err) => {
+        console.log('Something went wrong!', err);
+        res.send(err);
+      }
+    );
+});
+
+app.get('/scrape', async (req, res) => {
+  const spotifyId = req.query.spotifyId;
+  const { lastScraped } = await UserController.directFindUserBySpotifyId(spotifyId)
+  console.log('lastScraped', lastScraped)
   spotifyApi.setAccessToken(req.query.token);
   spotifyApi
     .getMyRecentlyPlayedTracks({
-      limit: 10,
+      limit: 50,
+      after: lastScraped || 0
     })
     .then(
-      (data) => {
+      async (data) => {
+        const ids = {}
+        data.body.items.forEach(item => {
+          const id = `listenStats.${item.track.id}`
+          if(ids[id]) {
+            ids[id] = ids[id] + 1;
+          } else {
+            ids[id] = 1
+          }
+        })
+        console.log(ids)
+        const newLastScraped = data.body.cursors?.after || lastScraped
```

Comment:
```suggestion
        const newLastScraped = data.body.cursors?.after || lastScraped || 0
```
Right now, I get `lastScraped = undefined` and newLastScraped is also undefined